controllers/machinedeployment_controller.go

```diff
@@ -163,6 +135,36 @@ func (r *MachineDeploymentReconciler) reconcile(ctx context.Context, d *clusterv
 		}
 	}
 
+	// Test for an empty LabelSelector and short circuit if that is the case
+	// TODO: When we have validation webhooks, we should likely reject on an empty LabelSelector
+	everything := metav1.LabelSelector{}
+	if reflect.DeepEqual(d.Spec.Selector, &everything) {
```

Comment:
I don't think we need this test any more because of line 154?

Reply:
If we don't check, it'd select all machinesets/machine in a cluster, which seems like a user error

Reply:
If we combine this with #1701, then I think we can completely remove user-defined label selectors.

Reply:
SGTM

Reply:
Ok, #1711 (replacement for #1701) has been merged. However, I don't think we should just remove user-defined label selection in a PR. If we want to do something like that, we should at least have an issue first.

Reply:
We can keep it as-is, until we have a proposal to remove the field hashed out